qsimcirq/qsim_simulator.py

```diff
@@ -213,19 +213,42 @@ def _run(
         """Run a simulation, mimicking quantum hardware.
 
         Args:
-            program: The circuit to simulate.
+            circuit: The circuit to simulate.
             param_resolver: Parameters to run with the program.
             repetitions: Number of times to repeat the run.
 
         Returns:
             A dictionary from measurement gate key to measurement
             results.
         """
+        self._check_for_confusion_matrix(circuit)
         param_resolver = param_resolver or cirq.ParamResolver({})
         solved_circuit = cirq.resolve_parameters(circuit, param_resolver)
 
         return self._sample_measure_results(solved_circuit, repetitions)
 
+    def _check_for_confusion_matrix(self, circuit: cirq.Circuit):
```

Comment:
I recommend moving this to `qsim_circuit.py` and calling it in `QSimCircuit.translate_cirq_to_qsim` instead. Every Cirq circuit that gets converted to qsim must go through that method, so putting it there would save having to call it from every simulation method in this file.

Reply:
I moved the method to QSimCircuit and updated the unit test accordingly. However, I want to note the  invocation pattern seen here: https://github.com/quantumlib/Cirq/issues/6305 does not result in an exception being thrown.

Reply:
> the invocation pattern seen here: [quantumlib/Cirq#6305](https://github.com/quantumlib/Cirq/issues/6305) does not result in an exception being thrown.

That's deeply surprising to me. Could you try adding the check in `translate_cirq_to_qtrajectory` as well? https://github.com/quantumlib/qsim/blob/401ebec99628050d6e41cc43925027966f630ebd/qsimcirq/qsim_circuit.py#L438

Reply:
I added the check here as well and the `ValueError` still wasn't raised. Your surprise makes me want to make sure my environment doesn't have anything odd/wonky. Let me get a summary of my env (python version qsim ver, cirq ver) and follow up

Reply:
Python version: 3.9
-----
cirq==1.2.0
qsimcirq==0.16.3

Reply:
How was this resolved?

Reply:
I confirmed that there was nothing wonky with my setup or library versions and the indeed in neither `translate_cirq_to_qsim` or `translate_cirq_to_qtrajectory` when the circuit is executed like so

`qsimcirq.QSimSimulator(seed=0).run(circuit, repetitions=100)`